### PR TITLE
document barcode scanner with sample code

### PR DESCRIPTION
**Pull request by sheykerkup**
addresses odk-x/tool-suite-X#119

#### What is included in this PR?
This PR contains sample code and steps to scan a barcode from ODK-X Tables application.

#### What problems did you encounter?
Could not launch intent when I just started implementation but after debugging noticed I had to install 
a barcode scanner application with the `com.google.zxing.client.android.SCAN` package.

### CODE REVIEW

**Comment by sheykerkup** [comment]
@wbrunette @elmps2018  @EstherPhebe  this is the complete process.

File: odkx-src/tables-intents.rst [hunk 1]
```diff
@@ -0,0 +1,243 @@
...
+.. _tables-intents:
+
+.. _tables-intents-sample-intents:
+
+This guide is for creating and launching intents supported by ODK-X Tables using `odkCommonIf` Java Interface through the `odkCommon.doAction` call. 
+
+Barcode Scanner
+~~~~~~~~~~~~~~~~~~~
+
+This section describes the process of creating a simple Tables app capable of scanning a barcode and displaying the results in a custom webpage.
```
Comment (elmps2018) [inline odkx-src/tables-intents.rst:16]: Minor style edit: helpful to say ODK-X Tables

File: odkx-src/tables-intents.rst [hunk 2]
```diff
@@ -0,0 +1,243 @@
...
+In this example, we will make use of the `Census Form` created earlier in the :ref:`build-app` section. 
+
+On :guilabel:`Application Designer`, open :file:`app/config/tables/census/html/census_list.html` and add a button to scan a barcode.
```
Comment (elmps2018) [inline odkx-src/tables-intents.rst:19]: In not On

File: odkx-src/tables-intents.rst [hunk 3]
```diff
@@ -0,0 +1,243 @@
...
+                <ul id="list"></ul>
+            </div>
+            <script>
+                $(function() { resumeFn(0); });
+            </script>
+        </body>
+    </html>
+
+
+Set up an on-click listener to the button to call `odkCommon.doAction`. 3 parameters will be passed to this function.
```
Comment (elmps2018) [inline odkx-src/tables-intents.rst:67]: "3 parameters" please write out numbers when starting a sentence or when less than ten ("Three parameters")

File: odkx-src/tables-intents.rst [hunk 4]
```diff
@@ -0,0 +1,243 @@
...
+
+
+Set up an on-click listener to the button to call `odkCommon.doAction`. 3 parameters will be passed to this function. 
+
+#. dispatchStruct: holds reconstructive state for JS. This is always sent back to the page. 
+#. action: `SCAN` in our case :code:`com.google.zxing.client.android.SCAN`.
+#. intent object: Intent extras - use :code:`null`.
+
+The next step is to register a listener. A listener listens for updates in a queue and triggers a callback function to process these updates. 
+In this case, the scan will be launched in a different app and to use the data returned in our Tables application, a callback function needs to be defined.
```
Comment (elmps2018) [inline odkx-src/tables-intents.rst:74]: ODK-X Tables

File: odkx-src/tables-intents.rst [hunk 5]
```diff
@@ -0,0 +1,243 @@
...
+The registered listener will then trigger this callback function whenever there is an update to be processed. 
+Whatever has to be done with the result of the scan can be configured in the callback function.
+In this case we just log the output to the console and print the results to the web page.
```
Comment (elmps2018) [inline odkx-src/tables-intents.rst:77]: web page here is confusing since not a web page. Screen maybe?

File: odkx-src/tables-intents.rst [hunk 6]
```diff
@@ -0,0 +1,243 @@
...
+        if (i < barcodeResultSet.getCount()) {
+            setTimeout(resumeFn, 0, i);
+        }
+    };
+
+    var cbFailure = function(error) {
+        console.log('barcode getViewData CB error : ' + error);
+    };
+
+Deploy changes to the device. The Survey form will not be a requirement for this test since we earlier hooked up the button in the list view. Select the census form and click on the button to launch a barcode. This launches the barcode reader enabling you to capture the information on the barcode. Immediately this is done, control switches back to the activity responsible for launching the intent, presenting you with the results from the scan. The results are now available for further processing as desired.
```
Comment (elmps2018) [inline odkx-src/tables-intents.rst:216]: "Immediately this is done" should be "Immediately after this is done"

File: odkx-src/tables-intents.rst [hunk 7]
```diff
@@ -0,0 +1,243 @@
...
+        "jsonValue": {
+            "status": -1,
+            "result": {
+                "SCAN_RESULT": "6801315196662",
+                "SCAN_RESULT_FORMAT": "EAN_13"
+            }
+        }
+    }
+
+Sample Screenshots are shown below:
```
Comment (elmps2018) [inline odkx-src/tables-intents.rst:234]: screenshots should be lower case.

File: odkx-src/tables-intents.rst [hunk 8]
```diff
@@ -0,0 +1,243 @@
...
+
+.. image:: /img/tables-sample-app/tables-barcode-button.png
+  :alt: ODK-X Tables barcode scan before
+
+.. image:: /img/tables-sample-app/tables-barcode-results.png
+  :alt: ODK-X Tables barcode scan results
+
+.. note::
+    If clicking on the button does not launch any activity to scan a barcode, it is possible that no barcode reader application exists one the device or the barcode with the zing package. Install a `zxing scanner` for example, :program:`QR & Barcode Reader` from TeaCapps.
```
Comment (elmps2018) [inline odkx-src/tables-intents.rst:243]: Is zing package right?

**Review by elmps2018 — CHANGES_REQUESTED** [review]
@sheykerkup thanks for working on this! I reviewed the explanation. @EstherPhebe can you review the tables code parts or should @linl33 or @wbrunette or @jbeorse check that?

File: odkx-src/tables-intents.rst (hunk 8 above)
Comment (sheykerkup) [reply odkx-src/tables-intents.rst:243]: `com.google.zxing.client.android.SCAN` is the intent so zxing package. Thanks

**Comment by sheykerkup** [comment]
@elmps2018  thanks for the review will fix them soon.

**Comment by sheykerkup** [comment]
@elmps2018  I am done with these changes. Thanks.

**Review by elmps2018 — COMMENTED** [review]
These look good to me but definitely want @linl33 or @wbrunette or @EstherPhebe or @jbeorse to check the tables piece!

**Comment by elmps2018** [comment]
@sheykerkup one of the checks failed, you should be able to see the issue in the details.

**Comment by sheykerkup** [comment]
> @sheykerkup one of the checks failed, you should be able to see the issue in the details.

passing now.

**Review by elmps2018 — APPROVED** [review]

**Pull request merged**